tasks/config.yml

```diff
@@ -23,6 +31,34 @@
   notify: Restart systemd-networkd
   register: systemd_networkd_status
 
+- name: Create systemd conf folder
+  become: true
+  ansible.builtin.file:
+    path: "{{ systemd_networkd_conf_directory }}"
+    state: directory
+    mode: 0755
+    owner: root
+    group: root
+
+- name: Systemd configuration files exist
+  become: true
+  vars:
+    all_files:
+      - type: conf
+        files: "{{ systemd_networkd_conf | dict2items }}"
+  template:
+    src: "systemd_networkd_config.j2"
+    dest: "{{ systemd_networkd_conf_directory }}/{{ item.1.key }}.{{ item.0.type }}"
+    owner: root
+    group: systemd-network
+    mode: 0640
+  loop: "{{ query('subelements', all_files, 'files' ) }}"
+  loop_control:
+    label:
+      type: "{{ item.0.type }}"
+      name: "{{ item.1.key }}"
+  notify: restart systemd-networkd
```

Comment:
I don't understand this part. Why don't you make `/etc/systemd/network` a variable, from this block line 14:

```
  ansible.builtin.template:
    src: systemd_networkd_config.j2
    dest: /etc/systemd/network/{{ item.1.key }}.{{ item.0.type }}
```

Reply:
You are right, I don't remember why I did it that way, and I don't see the point having a separate task, let me change it ASAP.